### ISSUE TITLE
dune build errors

### ISSUE THREAD

**Issue by OlivierNicole** [jmid/multicoretests#3]
I'm trying to use the `lin` library in a project, but `dune build` is failing on current main (a7a032f at the time of writing):

```
File "lib/STM.ml", line 308, characters 23-32:
308 |     Test.make ~retries:rep_count ~count ~name:("parallel " ^ name ^ " (w/shrink retries)")
                             ^^^^^^^^^
Error: The function applied to this argument has type
         ?if_assumptions_fail:[ `Fatal | `Warning ] * float ->
         ?long_factor:int ->
         ?max_gen:int ->
         ?max_fail:int -> ?small:('a -> int) -> QCheck2.Test.t
This argument cannot be applied with label ~retries
File "lib/lin.ml", line 190, characters 30-31:
190 |     Test.make ~count ~retries:3 ~name:("Linearizable " ^ name)
                                    ^
Error: The function applied to this argument has type
         ?if_assumptions_fail:[ `Fatal | `Warning ] * float ->
         ?long_factor:int ->
         ?max_gen:int ->
         ?max_fail:int -> ?small:('a -> int) -> QCheck2.Test.t
This argument cannot be applied with label ~retries
File "/home/olivier/compiler/trunk/_opam/lib/lockfree/dune-package", line 1, characters 11-14:
1 | (lang dune 3.0)
               ^^^
Error: Version 3.0 of the dune language is not supported.
Supported versions of this extension in version 3.0 of the dune language:
- 1.0 to 1.12
- 2.0 to 2.9
```

I am currently looking for the causes.

**Comment by jmid** [comment]
With a recent trunk ocaml installed building requires something like the two first steps from the CI:
https://github.com/jmid/multicoretests/blob/483fa6415d2d8c12c92111d6d5218f2457d5667e/.github/workflows/linux-500-workflow.yml#L27-L29

This is not clear from the outside though! To help others it would be nice to write up "Installation instructions" in the README.

The dependencies include
- a number of pin-depends in the opam-file pointing to packages patched for 5.0
- a pin to the QCheck repo - since we use the `~retries` feature which has been merged  - https://github.com/c-cube/qcheck/pull/212 but not released yet

IIUC, part of the above error message complains about not knowing about the new `~retries` parameter.

**Comment by OlivierNicole** [comment]
Indeed, `opam install . --deps-only` allowed me to `dune build` without errors.

 > This is not clear from the outside though! To help others it would be nice to write up "Installation instructions" in the README.

Done in #6.
